cirq/google/calibration/phased_fsim.py

```diff
@@ -130,3 +140,217 @@ def override_by(self, other: 'PhasedFSimCharacterization') -> 'PhasedFSimCharact
+        """Magic method for the JSON serialization protocol."""
+        return {
+            'cirq_type': 'PhasedFSimCalibrationResult',
+            'gate': self.gate,
+            'parameters': [(key[0], key[1], self.parameters[key]) for key in self.parameters],
+        }
+
+
+# We have to relax a mypy constraint, see https://github.com/python/mypy/issues/5374
+@dataclasses.dataclass(frozen=True)  # type: ignore
+class PhasedFSimCalibrationRequest(abc.ABC):
+    """Description of the request to characterize PhasedFSimGate.
+
+    Attributes:
+        pairs: Set of qubit pairs to characterize. A single qubit can appear on at most one pair in
+            the set.
+        gate: Gate to characterize for each qubit pair from pairs. This must be a supported gate
+            which can be described cirq.PhasedFSim gate.
+    """
+
+    # TODO: Validate that each pair is unique and non-overlaping with any other pair.
+    pairs: Tuple[Tuple[Qid, Qid], ...]
+    gate: Gate  # Any gate which can be described by cirq.PhasedFSim
+
+    @abc.abstractmethod
+    def to_calibration_layer(self) -> CalibrationLayer:
+        """Encodes this characterization request in a CalibrationLayer object."""
+
+    @abc.abstractmethod
+    def parse_result(self, result: CalibrationResult) -> PhasedFSimCalibrationResult:
+        """Decodes the characterization result issued for this request."""
+
+
+@json_serializable_dataclass(frozen=True)
+class FloquetPhasedFSimCalibrationOptions:
+    """Options specific to Floquet PhasedFSimCalibration.
+
+    Some angles require another angle to be characterized first so result might have more angles
+    characterized than requested here.
+
+    Attributes:
+        characterize_theta: Whether to characterize θ angle.
+        characterize_zeta: Whether to characterize ζ angle.
+        characterize_chi: Whether to characterize χ angle.
+        characterize_gamma: Whether to characterize γ angle.
+        characterize_phi: Whether to characterize φ angle.
+    """
+
+    characterize_theta: bool
+    characterize_zeta: bool
+    characterize_chi: bool
+    characterize_gamma: bool
+    characterize_phi: bool
+
+
+@dataclasses.dataclass(frozen=True)
+class FloquetPhasedFSimCalibrationResult(PhasedFSimCalibrationResult):
+    """PhasedFSim characterization result specific to Floquet calibration.
+
+    Attributes:
+        options: Options of the characterization from the request.
+    """
+
+    options: FloquetPhasedFSimCalibrationOptions
+
+    @classmethod
+    def _from_json_dict_(
+        cls,
+        parameters: List[Tuple[Qid, Qid, PhasedFSimCharacterization]],
+        gate: Gate,
+        **kwargs,
+    ) -> 'PhasedFSimCalibrationResult':
+        """Magic method for the JSON serialization protocol.
+
+        Converts serialized dictionary into a dict suitable for
+        class instantiation."""
+        return cls(cls._create_parameters_dict(parameters), gate, kwargs['options'])
```

Comment:
can't you change the superclass's implementation of `_from_json_dict_` to pass all extra `kwargs` to the `cls()` constructor call? Then you wouldn't have to override it here.

One thing to note: You probably have to pop `'cirq_type'` from `kwargs` first.

Reply:
Done